app/src/main/java/com/kylecorry/trail_sense/main/MainActivity.kt

```diff
@@ -131,6 +131,40 @@ class MainActivity : AndromedaActivity() {
         }
     }
 
+    /**
+     * Setup Statusbar and Bottom Nav color based on theme selected
+     */
+    private fun setBarsColorBasedOnCurrentTheme(theme: UserPreferences.Theme){
+        when(theme){
+            UserPreferences.Theme.Black, UserPreferences.Theme.Night -> {
+                window.apply {
+                    statusBarColor = Color.BLACK
+                    decorView.rootView.setBackgroundColor(Color.BLACK)
+                }
+                bottomNavigation.setBackgroundColor(Color.BLACK)
+            }
+            UserPreferences.Theme.Dark -> {
```

Comment:
I believe all of these (excluding Black and Night) can be combined under either `isDarkThemeOn()` and `!isDarkThemeOn()` (or just `else`). That should simplify this and avoid any potential discrepancies in sunriseSunsetTheme (it would be very rare for that to happen though).

Ex. 

when {
theme == UserPreferences.Theme.Black || theme == UserPreferences.Theme.Night -> 
// Same as today
}
isDarkThemeOn() -> {
window.statusBarColor = ColorUtils.backgroundColor(window.decorView.rootView.context)
}
else -> {
// Light
window.statusBarColor = ColorUtils.backgroundColor(window.decorView.rootView.context)
WindowInsetsControllerCompat(window, window.decorView).isAppearanceLightStatusBars = true
}

Reply:
Yes, the code is simpler that way. Fixed.